### PR TITLE
Add dotnet20

### PR DESCRIPTION
**Pull request by Zemogiter**
Help needed.
Despite copying what was done at winetricks the installer for dotnet20 throws this error:

zrzut ekranu z 2018-12-17 23-03-25

### CODE REVIEW

**Comment by qparis** [comment]
When does the error occur?

**Comment by Zemogiter** [comment]
@qparis at `this.run(setupFile, [setupFile, "/q:a", "/c:\"install.exe /q\""], null, false, true);`

**Comment by qparis** [comment]
Try to omit the quotes in the third argument

**Comment by Zemogiter** [comment]
No it doesn't.

**Comment by qparis** [comment]
Same error?

**Comment by qparis** [comment]
> No it doesn't.

qpmac:~ qparis$ cat test.py 
import sys

print(sys.argv)
qpmac:~ qparis$ python test.py /q /c:"install.exe /q"
['test.py', '/q', '/c:install.exe /q']

This should work pretty much like this

File: Engines/Wine/Verbs/dotnet20/script.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+include(["engines", "wine", "verbs", "luna"]);
+include(["utils", "functions", "filesystem", "files"]);
+include(["engines", "wine", "plugins", "windows_version"]);
+
+/**
+* Verb to install .NET 2.0
+* @returns {Wine} Wine object
+*/
+Wine.prototype.dotnet20 = function () {
+    if (this.architecture() = "x86") {
```
Comment (qparis) [inline Engines/Wine/Verbs/dotnet20/script.js]: Be careful, you forgot the == (instead of a simple =)

File: Engines/Wine/Verbs/dotnet20/script.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            .name("NetFx64.exe")
+            .get();
+    }
+
+    this.uninstall("Mono");
+
+    this.wizard().wait(tr("Please wait while {0} is installed ...", ".NET Framework 2.0"));
+    this.run(setupFile, [setupFile, "/q:a", "/c:install.exe /q"], null, false, true);
+
+    if (this.architecture() = "x86"){
```
Comment (qparis) [inline Engines/Wine/Verbs/dotnet20/script.js]: Same here

**Comment by Zemogiter** [comment]
Now the errors are gone aside from the on from screenshot.

**Comment by qparis** [comment]
What if you try:
    this.run(setupFile, [setupFile, "/q", "/c:install.exe /q"], null, false, true);

**Comment by qparis** [comment]
Ok, I understand the problem now.

wine.run("dotnetfx.exe", ["/q:a", "/c:install.exe /q"], null, false, true)

instead of 

wine.run("dotnetfx.exe", ["dotnetfx.exe", "/q:a", "/c:install.exe /q"], null, false, true)

File: Engines/Wine/Verbs/dotnet20/script.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            .get();
+    }
+
+    this.uninstall("Mono");
+
+    this.wizard().wait(tr("Please wait while {0} is installed ...", ".NET Framework 2.0"));
+    this.run(setupFile, [setupFile, "/q:a", "/c:install.exe /q"], null, false, true);
+
+    if (this.architecture() == "x86"){
+        remove(this.system32directory() + "/msvc" + "?" + "80.dll");
```
Comment (qparis) [inline Engines/Wine/Verbs/dotnet20/script.js]: I'm afraid that this won't work. We need to enumerate all the files

**Comment by Zemogiter** [comment]
Just tested this verb in my Sims 3 script. The game now launches. Verb is fully functional.

**Pull request merged**